### ISSUE TITLE
[Python] xbmc.getRegion('time') returns wrong scheme if language is set to German

### ISSUE THREAD

**Issue by b-jesch** [xbmc/xbmc#23974]
### Describe the bug
xbmc.getRegion('time') returns '%H:mm:ss' instead of the correct scheme '%H:%M:%S' when language settings is set to "German". 

From an addon that calculate some time expression, the xbmc.getRegion('time') is processed and logged. For a german language setting xbmc.getRegion('time') returns '%H:mm:ss' instead of '%H:%M:%S'. As a result time calculations with `<dateobject>.strftime()` will fail. If region is set to english, all is ok (returns '%I:%M:%S %p').

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Until Nexus the time scheme calculation is ok. Please revert or correct this.

## Actual Behavior

xbmc.getRegion('time') returns wrong scheme if german language is selected.

### To Reproduce

1. set Kodi to german language
2. create an addon or script that logged xbmc.getRegion('time')

### Debuglog

The debuglog can be found here: There's no complete debug.log as the following snippet describes the problem:
```
2023-10-21 16:24:11.804 T:4153     info <general>: [script.knftv.announcer 3.1.6+matrix] Local time format of client: %d.%m.%Y %H:mm:ss
2023-10-21 16:24:11.830 T:4153    error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'ValueError'>
                                                   Error Contents: time data '21.10.2023 20:15' does not match format '%d.%m.%Y %H:mm:ss'
```

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: tested on CoreElec, Ubuntu 20.04
 - Kodi version: 21 alpha2, still remains on beta1 (reported on kodinerds forum: https://www.kodinerds.net/thread/68354-filmtipps-im-freien-tv-anstehende-sendetermine/?postID=731770#post731770)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DeltaMikeCharlie** [comment]
@b-jesch - In the log snippet that you provided, the ‘time data’ is lacking seconds ('21.10.2023 20:15') but the format included seconds ('%d.%m.%Y %H:mm:ss').  Could the error be fixed by concatenating “:00” to your ‘time data’ prior to conversion?

Also, this issue sounds similar to a PR regarding date format: https://github.com/xbmc/xbmc/pull/22982

What ‘German’ locale are you using and what does your time format look like in the langinfo.xml of your language pack?  When I look in the latest DE language pack, all time formats are “HH:mm:ss”.  This should translate to “%H:%M:%S” as you say.

https://github.com/xbmc/xbmc/blob/cb947dde2c91c26b38eb83459deb378f68c3a4b1/xbmc/interfaces/legacy/ModuleXbmc.cpp#L439

You say that when your language is set to English, it works OK.  There are multiple ‘English’ time formats depending upon the region selected.  Exactly which ‘English’ locale are you using with your test?  The format “%I:%M:%S %p” suggests an original format of “h:mm:ss xx”.  This is strange because your input data would still lack seconds.

**Comment by b-jesch** [comment]
From the langinfo.xml:
```
<region name="Deutschland" locale="DE">
<dateshort>DD.MM.YYYY</dateshort>
<datelong>DDDD, D. MMMM YYYY</datelong>
<time symbolAM="" symbolPM="">HH:mm:ss</time>
<tempunit>C</tempunit>
<speedunit>kmh</speedunit>
<timezone>CEST</timezone>
</region>
```

All time formats are set to "regional", in case of English see screenshot.
![Bildschirmfoto von 2023-10-26 09-16-26](https://github.com/xbmc/xbmc/assets/10152384/0446331b-0236-4928-b343-9223ffe0202e)

**Comment by DeltaMikeCharlie** [comment]
Have you tried `xbmc.getRegion('timeraw')` to see if that matches your language file?  It may give a hint for debugging.

**Comment by b-jesch** [comment]
Just tried, returns the same false scheme (%H:mm:ss) on a CoreElec 21.0beta1 compiled at 20.10.2023. Btw. the parameter 'timeraw' is not documented yet: https://xbmc.github.io/docs.kodi.tv/master/kodi-base/d6/db2/group__python__xbmc.html#ga04efad52a8903be517a8310db6a8505f

On a Linux (Ubuntu Focal) system with Kodi 21.0 alpha compiled at 04.05.2023 the scheme returns the correct value (%H:%M:%S), but timeraw returns an empty string.

**Comment by DeltaMikeCharlie** [comment]
I thought that the issue sounded similar to something that I had seen before.  I'm out of ideas.  Sorry.

**Comment by mglae** [comment]
In case GttHub did not notify: fix is in #24009

**Comment by b-jesch** [comment]
Thanks for info. I will close this issue when the fix has merged.
